core/src/chain/mod.rs

```diff
@@ -34,14 +34,16 @@ impl<T: HashTable> Iterator for ChainIterator<T> {
     fn next(&mut self) -> Option<Pair> {
         let previous = self.current.take();
         self.current = previous.as_ref()
-                        .and_then(|p| p.header().next())
+                        .and_then(|p| p.header().prev())
```

Comment:
this does bother me, putting `prev` in a `next` call is confusing :/

maybe `link` or `header_link`, which avoids directionality

Reply:
Changed to `link`.
I like it because it fits the Chain metaphore.
I would recommend we stick to using `link` as a noun and not a verb in all the API, otherwise it might cause confusion.

Reply:
This is fine, the only issue here is that it's a slightly different nomenclature to what we used in the go version which was: 
```
	HeaderLink Hash // link to previous header
	EntryLink  Hash // link to entry
	TypeLink   Hash // link to header of previous header of this type
```
But I think this is fine, and given the context won't cause confusion.

Reply:
@zippy i expect we'll still use that nomenclature, but it would land in the serde config rather than rust methods right?

Reply:
@zippy I wasn't aware of that naming in Go, and I would have indeed proposed that instead. However I'm wondering how much do we want to stick to the Go names?
I think Go names are a starting point, and than we make use of the opportunity to improve them if we see a clear possible improvement based on our experience with the Go code.
Since I don't actually have experience with the Go code, I'm looking at it with new/fresh eyes.

Reply:
@ddd-mtl @zippy we did actually discuss this exact thing in a meeting a while back and agreed that breaking changes are OK here

that said, i agree it makes sense to use what we already have as a starting point

there's this old thing from june https://github.com/holochain/holochain-rust/issues/75